File: DSA/Ansh-Gulati/Day-2/Day-2_Question-1.cpp
#include <iostream>
using namespace std;

bool classCancelledOrNot(int size, int arrival_times[], int threshold)
{
    int count = 0;

    for (int i = 0; i < size; i++)
    {
        if (arrival_times[i] < 0)
        {
            count++;
        }
    }

    if (count < threshold)
    {
        return 1;
    }
    else
    {
        return 0;
    }
}

int main()
{
    int size;
    cout << "Enter the size of the array: ";
    cin >> size;

    int arrival_times[100];

    cout << "Input the arrival times in the array: ";

    for (int i = 0; i < size; i++)
    {
        cin >> arrival_times[i];
    }

    int threshold;
    cout << "Enter the threshold number of students : ";
    cin >> threshold;

    if (classCancelledOrNot(size, arrival_times, threshold))
    {
        cout << "YES" << endl;
    }
    else
    {
        cout << "NO" << endl;
    }
}

File: DSA/Rakshit-Singh/Day 2/Day_2_Question_1.java
class Day_2_Question_1
{
    public String Class(int threshold,int[] arrival_times){
        int count=0;
        for(int x=0;x<arrival_times.length;x++){
            if(arrival_times[x]<0){
                count+=arrival_times[x]*-1;
            }
        }
        if(count>=threshold){
            return "YES";
        }
        return "NO";
    }
}